.github/workflows/publish_pytorch_dev_docker.yml

```diff
@@ -71,4 +72,89 @@ jobs:
+          labels: |
+            org.opencontainers.image.title=PyTorch ROCm Dev Image
+            org.opencontainers.image.description=Nightly ROCm PyTorch Dev Docker for target ${{ matrix.targets.amdgpu_target }}
+            org.opencontainers.image.version=${{ env.VERSION }}
+            org.opencontainers.image.created=${{ steps.meta.outputs.created }}
+            org.opencontainers.image.revision=${{ github.sha }}
+
+      - name: Export digest
+        id: export
+        run: |
+          digest="${{ steps.build_push.outputs.digest }}"
+          case "${{ matrix.targets.amdgpu_target }}" in
+            gfx942) echo "gfx942=$digest" >> $GITHUB_OUTPUT ;;
+          esac
+          echo "Exported digest: $digest for ${{ matrix.targets.amdgpu_target }}"
+
+  verify-image:
+      needs: build-and-push-image
+      strategy:
+        matrix:
+          include:
+            - amdgpu_target: gfx942
+              image_digest: ${{ needs.build-and-push-image.outputs.gfx942_digest }}
+      uses: ./.github/workflows/verify_docker_image.yml
+      with:
+        amdgpu_target: ${{ matrix.amdgpu_target }}
+        image_digest: ${{ matrix.image_digest }}
+
+  on-success:
+    needs: [build-and-push-image, verify-image]
+    if: ${{ success() }}
+    strategy:
+      matrix:
+        include:
+          - amdgpu_target: gfx942
+            image_digest: ${{ needs.build-and-push-image.outputs.gfx942_digest }}
+    runs-on: azure-linux-scale-rocm
+    steps:
+      - name: Set version from build job
+        run: |
+          echo "VERSION=${{ needs.build-and-push-image.outputs.version }}" >> "$GITHUB_ENV"
+          echo "Using version: ${{ needs.build-and-push-image.outputs.version }}"
+
+      - name: Log in to registry
+        uses: docker/login-action@9780b0c442fbb1117ed29e0efdff1e18412f7567 # v3.3.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Write pinned version to file
+        run: echo "${{ env.VERSION }}" > latest_verified_image_version_pinned.txt
+
+      - name: Upload pinned version file
+        uses: actions/upload-artifact@4cec3d8aa04e39d1a68397de0c4cd6fb9dce8ec1 # v4.6.1
+        with:
+          name: latest_verified_image_version_pinned-${{ matrix.amdgpu_target }}
+          path: latest_verified_image_version_pinned.txt
+
+      - name: Tag and push verified image
+        run: |
+          IMAGE=ghcr.io/rocm/therock_pytorch_dev_ubuntu_24_04_${{ matrix.amdgpu_target }}
+          DIGEST=${{ matrix.image_digest }}
+          VERSION=${{ env.VERSION }}
+
+          echo "Using image: $IMAGE"
+          echo "Using digest: $DIGEST"
+          echo "Using version: $VERSION"
+
+          docker pull $IMAGE@$DIGEST
+          docker tag $IMAGE@$DIGEST $IMAGE:verified
+          docker tag $IMAGE@$DIGEST $IMAGE:$VERSION
+          docker push $IMAGE:verified
+          docker push $IMAGE:$VERSION
```

Comment:
I guess you rather want to use `docker/build-push-action` (https://docs.docker.com/build/ci/github-actions/manage-tags-labels/).